pkgs/development/libraries/lucene++/default.nix

```diff
@@ -1,28 +1,36 @@
-{ lib, stdenv, fetchFromGitHub, cmake, boost, gtest }:
+{ lib, stdenv, fetchFromGitHub, cmake, boost, gtest, zlib }:
 
 stdenv.mkDerivation rec {
   pname = "lucene++";
-  version = "3.0.7";
+  version = "3.0.8";
 
   src = fetchFromGitHub {
     owner = "luceneplusplus";
     repo = "LucenePlusPlus";
     rev = "rel_${version}";
-    sha256 = "06b37fly6l27zc6kbm93f6khfsv61w792j8xihfagpcm9cfz2zi1";
+    sha256 = "12v7r62f7pqh5h210pb74sfx6h70lj4pgfpva8ya2d55fn0qxrr2";
   };
 
+  nativeBuildInputs = [ cmake ];
+  buildInputs = [ boost gtest zlib ];
+
+  doCheck = true;
+
   postPatch = ''
-    sed -i -e '/Subversion *REQUIRED/d' \
-           -e '/include.*CMakeExternal/d' \
-           CMakeLists.txt
+     substituteInPlace src/test/CMakeLists.txt \
+            --replace "add_subdirectory(gtest)" ""
   '';
 
-  cmakeFlags = [ "-DGTEST_INCLUDE_DIR=${gtest}/include" ];
-  nativeBuildInputs = [ cmake ];
-  buildInputs = [ boost gtest ];
+  checkPhase= ''
+    runHook preCheck
+    LD_LIBRARY_PATH=$PWD/src/contrib:$PWD/src/core \
+            src/test/lucene++-tester
```

Comment:
Did they remove the test target? Maybe we can set LD_LIBRARY_PATH in preCheck?

Reply:
> Did they remove the test target? 

Yes, at least this particular target went away and I couldn't find another one. According to the README they want you to run this binary to run the tests. https://github.com/luceneplusplus/LucenePlusPlus/blob/rel_3.0.8/README.md#to-run-unit-test-suite

> Maybe we can set LD_LIBRARY_PATH in preCheck?

You mean like this ? :
```nix
preCheck = ''
  export LD_LIBRARY_PATH=$PWD/src/contrib:$PWD/src/core
'';
```

Reply:
> You mean like this ? :

Yeah but that is only useful when using test target.

Reply:
What was the last sentiment here? Do you want me to change it? I dont see much use in that.